Pages/Careers/CareersSearchPage.private.cs

```
@@ -0,0 +1,116 @@
+﻿using ATA_Dotnet_Selenium_task.Constants;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.WaitHelpers;
+
+namespace ATA_Dotnet_Selenium_task.Pages.Careers;
+
+internal partial class CareerSearchPage
+{
+    //3.	Write the name of any programming language in the field “Keywords” (should be taken from test parameter)
+    internal void EnterKeywordSearchCriteria(string searchText)
+    {
+        new WebDriverWait(_driver, GlobalVariables.ExplicitWaitDefault).
+            Until(ExpectedConditions.ElementToBeClickable(_keywordInput));
+        KeywordInput.Clear();
+        KeywordInput.SendKeys(searchText);
+
+        // Ignore search suggestions.
+        try
+        {
+            new WebDriverWait(_driver, GlobalVariables.ExplicitWaitShort).
+                   Until(ExpectedConditions.ElementIsVisible(_keywordSuggestions));
+            KeywordInput.SendKeys(Keys.Tab);
+        }
+        catch
+        {
+            // Continue with test execution.
+        }
+    }
+
+    //4.	Select “All Locations” in the “Location” field(should be taken from the test parameter)
+    internal void SelectLocationByText(string searchText)
+    {
+
+        new WebDriverWait(_driver, GlobalVariables.ExplicitWaitDefault).
+            Until(ExpectedConditions.ElementToBeClickable(_locationClickable));
+        LocationClickable.Click();
+        try
+        {
+            // Select by visible options (without searching).
+            var selection = LocationOptions.First(x => x.GetAttribute("title") == searchText);
+            selection.Click();
+        }
+        catch (Exception)
+        {
+            // Select by searching
+            LocationInput.SendKeys(searchText);
+            var selection = LocationOptions.First();
+            selection.Click();
+        }
+    }
```

Same thing with this method, we are doing more than one thing. Let's refactor this into two methods, one to click the dropdown, and the other to select the desired option.

Also, in the catch, let's log and throw the exceptions, let's figure out the approach to select options successfully and use that, let's never do workarounds inside a catch statement. Tests must be objective, factual, crude, and cold, if something fails, let's let it fail. But also, let's make sure our code works correctly all the time.

---

fixed